### PR TITLE
Prac 07 feedback

### PR DESCRIPTION
**Pull request by Jack-Jesse**
Code review please @rachel-chann

### CODE REVIEW

File: prac_07/project_management.py [hunk 1]
```diff
@@ -47,9 +47,10 @@ def main():
 
 def load_file(filename, projects):
     """Load file from file in to projects list."""
+    del projects[:]  # Delete previously loaded projects to avoid duplicates
```
Comment (ghost) [inline prac_07/project_management.py:50]: Is this the best way to handle this?

File: prac_07/project_management.py [hunk 2]
```diff
@@ -149,8 +150,14 @@ def add_new_project(projects):
...
-    date_string = input("Show projects that start after date (dd/mm/yy): ")
-    date = datetime.datetime.strptime(date_string, "%d/%m/%Y").date()
+    is_valid_date = False
+    while not is_valid_date:
+        try:
+            date_string = input("Show projects that start after date (dd/mm/yy): ")
+            date = datetime.datetime.strptime(date_string, "%d/%m/%Y").date()
+            is_valid_date = True
+        except ValueError:
+            print("Enter valid date format (dd/mm/yy)")
```
Comment (ghost) [inline prac_07/project_management.py:160]: It's a good idea to handle the value error. Would it be even better if you refactor these lines into a separate function?

File: prac_07/project_management.py [hunk 3]
```diff
@@ -95,8 +100,16 @@ def add_new_project(projects):
     projects.append(project)
```
Comment (ghost) [inline prac_07/project_management.py:147]: I thought this is the update_project function. Do you need to append the project?

File: prac_07/project_management.py [hunk 4]
```diff
@@ -95,8 +100,16 @@ def add_new_project(projects):
...
 
 
-def filter_projects_by_date():
-    print("Filter projects by date")
+def filter_projects_by_date(projects):
+    filtered_projects = []
```
Comment (ghost) [inline prac_07/project_management.py:152]: Well done! You may also consider using the list comprehension.

File: prac_07/project_management.py [hunk 5]
```diff
@@ -77,8 +77,18 @@ def update_project():
...
+    name = input("Name: ")
+    start_date = input("Start date: ")
+    priority = int(input("Priority: "))
+    cost_estimate = float(input("Cost estimate: "))
+    percent_complete = int(input("Percent complete: "))
+    project = Project(name, start_date, priority, cost_estimate, percent_complete)
+    if project.is_complete():
+        completed_projects.append(project)
+    else:
+        incomplete_projects.append(project)
```
Comment (ghost) [inline prac_07/project_management.py]: Do you need to save the projects in two lists?  You may consider doing the separation once in the display function.

File: prac_07/project_management.py [hunk 6]
```diff
@@ -43,12 +44,11 @@ def main():
...
-        in_file.readline()
+        in_file.readline()  # remove the first line
         for line in in_file:
-            projects.append(line)
-
-
-
+            parts = line.strip().split("\t")
+            project = Project(parts[0], parts[1], parts[2], parts[3], parts[4])
+            projects.append(project)
```
Comment (rachel-chann) [inline prac_07/project_management.py]: Well done!

Comment (ghost) [inline prac_07/project_management.py]: Well done!

File: prac_07/project_management.py [hunk 7]
```diff
@@ -43,22 +46,31 @@ def main():
...
-def load_file(filename, projects):
+def load_file(filename, incomplete_projects, completed_projects):
     with open(filename, "r", encoding="utf-8") as in_file:
         in_file.readline()  # remove the first line
         for line in in_file:
             parts = line.strip().split("\t")
-            project = Project(parts[0], parts[1], parts[2], float(parts[3]), parts[4])
-            projects.append(project)
+            project = Project(parts[0], parts[1], int(parts[2]), float(parts[3]), int(parts[4]))
+            if project.completion_percentage == 100:
```
Comment (ghost) [inline prac_07/project_management.py]: You may consider using the is_completed method you defined in the class.

**Review by ghost — COMMENTED** [review]
Nice work overall. I have made some suggestions on certain area.

**Pull request merged**